units/systemd-firstboot.service.in

```diff
@@ -14,7 +14,6 @@ DefaultDependencies=no
 Conflicts=shutdown.target
 After=systemd-remount-fs.service
 Before=systemd-sysusers.service sysinit.target shutdown.target
-ConditionPathIsReadWrite=/etc
```

Comment:
hmm, not sure i grok this. if /etc is read-only the tool is not going to be able to make any of its changes, no?

I mean, we generally create files atomically, i.e. via O_TMPFILE or random file names which are then moved into place, which means we need write access to /etc as a whole, a specific file underneath it does not suffice

Reply:
well if /etc/machine-id is a link to /var/machine-id, /etc can be read-only but we can still write to the machine-id file...

Reply:
but we generally create our files in /etc in an atomic fashion: i.e. create and rename. This means you cannot symlink stuff from /etc to elsewhere reasonably, as that would replace the symlink not the destination.

Reply:
/etc/machine-id is not initialized by systemd-firstboot.service btw, but by PID 1, much earlier.